### ISSUE TITLE
Doctrine_Core::HYDRATE_ON_DEMAND not working

### ISSUE THREAD

**Issue by fidelio314** [LExpress/doctrine1#20]
Hello, i found an issue, using the doctrine1 fork of LExpress with  symfony1.5

I have a pager built. With NbResults 600 (in an example)
then i user the method: 
` $oCustomers = $pager->getResults(Doctrine_Core::HYDRATE_ON_DEMAND);`
that gets us an object: **Doctrine_Collection_OnDemand**

then i have a while loop:

```
while ($oCustomer = $oCustomers->current()) {
.....
....
     $oCustomer->free();
     $oCustomers->next();
}
```

i would suppose to have 600 loops.

i actually have 2!

rolled back in symfony1.4 i confirm having 600 loops.

do you have any idea on the subject? 

thanks!

**Comment by bitak1** [comment]
I'm having the same issue after upgrade to LExpress/doctrine1 from original doctrine1, where following query was working just fine, unfortunately now HYDRATE_ON_DEMAND doesn't work anymore.
`        $uctuj_od_dt = "2015-10-01 00:00:00";
        $q_ucty = Doctrine_Query::create()
                ->select()
                ->from('Ucet')
                ->where('od >= ?', $uctuj_od_dt)
        ;

        $ucty = $q_ucty->execute(array(), Doctrine_Core::HYDRATE_ON_DEMAND);
//        $ucty = $q_ucty->execute();

        $counter = 0;
        foreach ($ucty_tel as $ucet) {
            // doing something
            $counter++;
        }`
If I use Doctrine_Core::HYDRATE_ON_DEMAND the $counter ends up to be only 2, instead of many hundreds, which are in database. If I use only $ucty = $q_ucty->execute(); then I get all the results as expected, but then it consumes to much memory on the server... I also noticed, that if I use query without the where clause, then $ucty = $q_ucty->execute(array(), Doctrine_Core::HYDRATE_ON_DEMAND); works fine.

**Comment by fruit** [comment]
The bug appeared after this fix: 150e1e4d11dcbde7064a9de98c3053ddffb8a0ff

@j0k3r Can you rollback it?

**Comment by j0k3r** [comment]
I'll let @jeromemacias decide. I've no idea why he did that.
Poke @miniche

**Comment by fruit** [comment]
I checked different behaviors... and...

`closeCursor()` method call should be removed from [line 108](/LExpress/doctrine1/blob/master/lib/Doctrine/Hydrator/Graph.php#L108) and [line 139](/LExpress/doctrine1/blob/master/lib/Doctrine/Hydrator/Graph.php#L139)

And `closeCursor()` can be left on [line 116](/LExpress/doctrine1/blob/master/lib/Doctrine/Hydrator/Graph.php#L116)

**Comment by fidelio314** [comment]
just tested the fix: i commented the two lines as done in the commit: a73ab58

`$stmt->closeCursor();`

(https://github.com/eFaqtBV/doctrine1/commit/a73ab5864b33a5e7fe29179998f5ee68baf28844)

it seems that works fine now.

Would you consider to merge it with the master branch?

**Comment by j0k3r** [comment]
I released [1.3.4](https://github.com/LExpress/doctrine1/releases/tag/v1.3.4) with that fix.
Thanks all and let us know if it works fine.
